changelog.lua

```diff
@@ -0,0 +1,374 @@
+local pl_template = require "pl.template"
+local pl_tablex = require "pl.tablex"
+local pl_file = require "pl.file"
+local pl_dir = require "pl.dir"
+local pl_path = require "pl.path"
+local pl_stringx = require "pl.stringx"
+local lyaml = require "lyaml"
+local pl_app = require 'pl.lapp'
+local cjson = require "cjson"
+local http = require("socket.http")
+
+local CHANGELOG_PATH -- absolute path of CHANGELOG directory
+do
+  local base_path = os.getenv("PWD")
+  local command = debug.getinfo(1, "S").source:sub(2)
+  local last_idx = pl_stringx.rfind(command, "/")
+  if last_idx then
+    base_path = pl_path.join(base_path, string.sub(command, 1, last_idx - 1))
+  end
+  CHANGELOG_PATH = base_path
+end
+local UNRELEASED = "unreleased"
+local REPOS = {
+  kong = "Kong/kong",
+}
+local JIRA_BASE_URL = "https://konghq.atlassian.net/browse/"
+local GITHUB_REFERENCE = {
+  pr = "https://github.com/%s/pull/%d",
+  issue = "https://github.com/%s/issues/%d"
+}
+local SCOPE_PRIORITY = { -- smallest on top
+  Performance = 10,
+  Configuration = 20,
+  Core = 30,
+  PDK = 40,
+  Plugin = 50,
+  ["Admin API"] = 60,
+  Clustering = 70,
+  Default = 100, -- default priority
+}
+
+setmetatable(SCOPE_PRIORITY, {
+  __index = function()
+    return rawget(SCOPE_PRIORITY, "Default") - 1
+  end
+})
+
+local function table_keys(t)
+  if type(t) ~= "table" then
+    return t
+  end
+  local keys = {}
+  for k, _ in pairs(t) do
+    table.insert(keys, k)
+  end
+  return keys
+end
+
+local function parse_github_ref(system, reference_type, references)
+  if references == nil or references == lyaml.null then
+    return nil
+  end
+  local parsed_references = {}
+  for i, ref in ipairs(references or {}) do
+    local repo = REPOS[system]
+    local ref_no = tonumber(ref) -- treat ref as number string first
+    local name = "#" .. ref
+    if not ref_no then -- ref is not a number string
+      local parts = pl_stringx.split(ref, ":")
+      repo = parts[1]
+      ref_no = parts[2]
+      name = pl_stringx.replace(tostring(ref), ":", " #")
+    end
```

Comment:
```suggestion
    end

```

Try to keep a consistent style to how `Kong/kong`'s code is written. It makes reading code easier too.